### ISSUE TITLE
lld-link does not support /time

### ISSUE THREAD

**Issue by JVApen** [llvm/llvm-project#99011]
Lld-link does not support [`/time`](https://learn.microsoft.com/en-us/cpp/build/reference/time-linker-time-information?view=msvc-170) as a command line option. The alternative `--time-trace` of lld ain't available either.
We've tried with `/mllvm`, though as `--time-trace` does not exist on llvm-level, it cannot be used either and a linker variant of `-Xclang` doesn't exist in lld-link.

When using `link.exe`, the `/time` gives:
````
  Wait PDB close: Total time = 0.000s
  Wait type merge: Total time = 0.000s
   Creating library library.lib and object library.exp
LibDef: Total time = 0.219s
  MD Merge: Total time = 0.015s
  Generate Transitions: Total time = 0.000s
  MD Finalize: Total time = 0.000s
Pass 1: Interval #1, time = 0.610s
  Wait PDB close: Total time = 0.000s
  Wait type merge: Total time = 0.000s
Pass 2: Interval #2, time = 0.031s
Final: Total time = 0.641s
````

Our internal monitoring of the link times has been broken by our switch from lld-link to link, which we did for performance reasons.
Without either of the 2 options, we are forced to use `link` in this monitoring, although we don't use that as a linker.
The information we currently extract from this are: the library name and the total time.
I believe that this information is also available in the `--time-trace` of lld, though I'm not sure about the library name.

**Comment by rnk** [comment]
LLD does support `/time`, and it produces similar information, although it is not designed to be format-compatible:
https://github.com/llvm/llvm-project/blob/main/lld/COFF/Options.td#L318
```
$ bin/lld-link -dll -... /time
  Input File Reading:               0 ms (  9.0%)
  GC:                               0 ms (  0.0%)
  ICF:                              0 ms (  0.2%)
  Code Layout:                      0 ms ( 20.3%)
  Commit Output File:               0 ms (  8.2%)
--------------------------------------------------
Total Linking Time:                 3 ms (100.0%)
```

I take your post to mean that lld-link is faster than link, which I suppose is always good to hear. :)

**Comment by rnk** [comment]
Maybe the next action here is to give the option help text so it shows up in help (`/?`) output.

**Comment by JVApen** [comment]
You are correct, we got to use the output it generated.

I'll keep this issue open to:
 - document /time (good starters issue)
 - checking if --time-trace can be added without much effort

**Comment by rnk** [comment]
It looks to me like `--time-trace` is available in since 356139bd027d65b6843cbd4eda642104cfe6cf8f / 2023, and I added help text for `/time` in 7aa6598a244d5b1197725f26d367fb51e6630415 . I think that's enough to close this, hopefully that gives you the tools you need.
